### PR TITLE
UHF-4905 hdbt documentation

### PR DESCRIPTION
**Pull request by teroelonen**
# [UHF-4905](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4905)

## What was done

* Documentation was updated to include more information.

## How to test

* [ ] Read the documentation and make sure that its easy to understand and doesn't include mistakes.

## Designers review

* [x] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

## Other PRs

* https://github.com/City-of-Helsinki/drupal-hdbt/pull/684
* https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/221
* https://github.com/City-of-Helsinki/drupal-helfi-platform/pull/135

[UHF-4905]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-4905?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,34 +33,51 @@ Explanations for commands.
 - `nvm use` : Install and use the correct version of Node.
 - `npm i` : As stated above; Install dependencies and link local packages.
 
+Related files.
+- `.nvmrc` : Defines the node version used to compile the theme.
+- `package.json and package-lock.json` : Defines the node modules for compiling the theme.
```
Comment (khalima) [inline README.md:38]: > - `package.json and package-lock.json` : Defines the node modules for compiling the theme.

Suggestion: 
- `package.json and package-lock.json` : Defines the node modules and scripts for compiling the theme.

File: README.md [hunk 2]
```diff
@@ -32,34 +33,51 @@ Explanations for commands.
...
+- `postcss.config.js and postcss.plugins.js` : Configurations for the postcss-tool that is run when the theme is built.
+You can read more about the tool here: https://postcss.org/
+- `webpack.config.js` : Configuration file for the webpack-tool that is used to actually build the theme. Similar tool
+to Gulp or Grunt. Usually if there is something wrong with the compiled theme files the culprit can be found here.
+- `webpack.svgToSprite.js` : This file is used to create a sprite of all the icons used on the site. It gets all the
+icons from the src-folder, compiles them into a sprite under `dist/icons` and also creates a css-file where the icons are
```
Comment (khalima) [inline README.md:44]: > It gets all the icons from the src-folder, compiles them into a sprite under `dist/icons` and...

Suggestion: 
It gets all the icons from the `./src`, compiles them into a sprite under `./dist/icons` and...

File: README.md [hunk 3]
```diff
@@ -32,34 +33,51 @@ Explanations for commands.
...
+referenced called _hdbt-icons.css_.
```
Comment (khalima) [inline README.md:45]: > referenced called _hdbt-icons.css_.

Suggestion: 
referenced called `./dist/css/hdbt-icons.css`.

**Review by Jussiles — APPROVED** [review]
Changes look good to me. Added one comment to HDBT admin theme. Same thing could be added to here too.

File: README.md [hunk 4]
```diff
@@ -32,34 +33,51 @@ Explanations for commands.
...
-│   └───icons
-│       |   sprite.svg
-│       └───subdir
-│           |   some-icon.svg
-└───dist
-    └───css
-    └───js
-    └───icons
-```
+### The config-folder
```
Comment (khalima) [inline README.md:49]: > The whole chapter `Structure for files and folders`

Suggestion: 
### The config folder
The config folder includes configurations that are used when installing a new project from scratch. These configuration files are copied under the `conf/cmi` folder and used there. Therefore, altering them under the theme doesn't change anything unless you are building a new instance.

### The dist and src folders
Under the `./src` folder, there are all the theme components that are being compiled to the `./dist` folder, such as CSS, JavaScript, icons, and fonts. The `./dist` folder includes the compiled version of the same information created using the commands listed under the commands title.

The theme styles under the `./scss` folder are structured by implementing principles from the ITCSS architecture but with small adjustments to make it work for the needs of the project.

### Templates folder
Under the `./templates` folder, the structure is similar to the base-theme stable9 with a few additions, such as the `./templates/module` folder that includes templates for the helfi-prefixed modules created for this project.

### Translations folder
The `./translations` folder includes translations for all the translatable strings provided by the hdbt-theme.

File: README.md [hunk 5]
```diff
@@ -138,7 +159,8 @@ aggregation from Drupal. Go to /admin/config/development/performance and uncheck
 site caches and you should be able to continue with your work.
 
 ### How can I add custom translations?
-Add your UI translations to ``./translations/{fi/sv}.po`` files like it is explained in Translation in Drupal 8 documentation: https://www.drupal.org/docs/understanding-drupal/translation-in-drupal-8.
+Add your UI translations to ``./translations/{fi/sv}.po`` files like it is explained in Translation in Drupal 8
```
Comment (khalima) [inline README.md:162]: > Add your UI translations to ``./translations/{fi/sv}.po`` files like it is explained in Translation in Drupal 8

Suggestion: 
Add your UI translations to `./translations/{fi/sv}.po` files like it is explained in Translation in Drupal

**Review by khalima — APPROVED** [review]
I think the documentation is good. However I do suggest few changes.. you can see them in comments.

File: README.md (hunk 1 above)
Comment (teroelonen) [reply README.md]: Fixed! 🦖

File: README.md (hunk 3 above)
Comment (teroelonen) [reply README.md]: Fixed! 🦖

File: README.md (hunk 4 above)
Comment (teroelonen) [reply README.md:49]: These were some good changes! Thank you! Fixed! 🦖

File: README.md (hunk 5 above)
Comment (teroelonen) [reply README.md]: Good catch 😄

File: README.md (hunk 2 above)
Comment (teroelonen) [reply README.md]: Fixed! 🦖

**Pull request merged**